### PR TITLE
Chore: add package needed for our precommit linting

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

The `clang` package that our conda environment installs doesn’t come with `clang-format-diff.py`, which is needed for our pre-commit linting hook. This adds the package that includes that file.

This [new package](https://anaconda.org/conda-forge/clang-tools) has the Apache 2.0 license.

## Reference

N/A

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question) 
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

**Review by birdsarah — CHANGES_REQUESTED** [review]
Can you update the Readme too?

**Comment by birdsarah** [comment]
You can set an environment variable in the env.yaml - https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#setting-environment-variables .... i'm just not sure how to reference the path of the environment.

**Comment by birdsarah** [comment]
For my last comment you don't have to update the PR, the README can include the manual step you mentioned. But I just thought I'd mention it.

File: env.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ channels:
   - conda-forge
 dependencies:
   - clang=14.0.6
+  - clang-tools=14.0.6
```
Comment (birdsarah) [inline env.yml:6]: And clang-format?? I don't know, but guessing yes.

Comment (mcleinman) [reply env.yml:6]: Not needed - clang-tools has a dependency on it, so it gets installed automatically:
```
clang-tools 14.0.6 default_hf5194b7_0
-------------------------------------
file name   : clang-tools-14.0.6-default_hf5194b7_0.conda
name        : clang-tools
version     : 14.0.6
build       : default_hf5194b7_0
build number: 0
size        : 9.8 MB
license     : Apache-2.0 WITH LLVM-exception
subdir      : osx-arm64
url         : https://repo.anaconda.com/pkgs/main/osx-arm64/clang-tools-14.0.6-default_hf5194b7_0.conda
md5         : 94cac2f8bc7b1579595d258bde3bde83
timestamp   : 2022-08-31 06:32:05 UTC
constraints : 
  - clangdev 14.0.6
dependencies: 
  - clang-format 14.0.6 default_hf5194b7_0
  - libclang >=14.0.6,<14.1.0a0
  - libclang-cpp14 >=14.0.6,<14.1.0a0
  - libcxx >=12.0.0
  - libllvm14 >=14.0.6,<14.1.0a0
```

**Comment by mcleinman** [comment]
Added readme info

**Comment by mcleinman** [comment]
> You can set an environment variable in the env.yaml - https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#setting-environment-variables .... i'm just not sure how to reference the path of the environment.

Same, and a quick search doesn't seem to get me there.

File: README.md [hunk 2]
```diff
@@ -35,8 +35,19 @@ Here are some useful links to start:
 * Localization happens on
   [Pontoon](https://pontoon.mozilla.org/projects/mozilla-vpn-client/).
 
+### Pre-commit formatting hook
+
 If you want to submit a pull-request, please, install the clang format
-pre-commit hook: `./scripts/git-pre-commit-format install`
+pre-commit hook that lints code.
```
Comment (bakulf) [inline README.md:41]: We need to write the pre-commit hook script. Am i wrong?

Comment (mcleinman) [reply README.md:41]: I looked at this PR, and understand what you meant. I neglected to add this as step 3 before. This has been fixed, thanks.

**Review by bakulf — APPROVED** [review]

**Review by birdsarah — APPROVED** [review]

**Pull request merged**